test/e2e/es/remote_cluster_test.go

```diff
@@ -69,7 +70,11 @@ func TestRemoteCluster(t *testing.T) {
 			test.Step{
 				Name: "Add some data to the first cluster",
 				Test: func(t *testing.T) {
-					require.NoError(t, elasticsearch.NewDataIntegrityCheck(k, es1Builder).Init())
+					dataIntegrityCheck := elasticsearch.NewDataIntegrityCheck(k, es1Builder)
+					if strings.HasPrefix(es1Builder.Elasticsearch.Spec.Version, "6") {
+						dataIntegrityCheck.WithSoftDeletesEnabled(true)
```

Comment:
I wonder whether we could just always explicitly enable soft deletes and avoid the conditional branch here? We could add a comment explaining that we are doing it because of older versions of Elasticsearch that did not have it enabled by default.